identity-android/src/main/java/com/android/identity/SessionEncryption.java

```diff
@@ -104,18 +128,59 @@ public SessionEncryptionReader(@NonNull PrivateKey eReaderKeyPrivate,
             byte[] info = "SKDevice".getBytes(UTF_8);
             byte[] derivedKey = Util.computeHkdf("HmacSha256", sharedSecret, salt, info, 32);
 
-            mSKDevice = new SecretKeySpec(derivedKey, "AES");
+            deviceSK = new SecretKeySpec(derivedKey, "AES");
 
             info = "SKReader".getBytes(UTF_8);
             derivedKey = Util.computeHkdf("HmacSha256", sharedSecret, salt, info, 32);
-            mSKReader = new SecretKeySpec(derivedKey, "AES");
+            readerSK = new SecretKeySpec(derivedKey, "AES");
         } catch (NoSuchAlgorithmException | InvalidKeyException e) {
             throw new IllegalStateException("Error deriving keys", e);
         }
+
+        if (mRole == ROLE_MDOC) {
+            mSKSender = deviceSK;
+            mSKReceiver = readerSK;
+        } else {
+            mSKSender = readerSK;
+            mSKReceiver = deviceSK;
+        }
     }
 
     /**
-     * Configure whether to send <code>SessionEstablishment</code> as the first message.
+     * Creates a new {@link SessionEncryption} object with the ROLE_MDOC_READER role.
+     *
+     * @param eReaderKeyPrivate the reader private ephemeral key.
+     * @param eReaderKeyPublic the reader public ephemeral key.
+     * @param eDeviceKeyPublic the device public key.
+     * @param encodedSessionTranscript the bytes of the <code>SessionTranscript</code> CBOR.
+     * @return a new {@link SessionEncryption} object with the ROLE_MDOC_READER role.
+     */
+    public static SessionEncryption createReaderSessionEncryption(@NonNull PrivateKey eReaderKeyPrivate,
+                                                           @NonNull PublicKey eReaderKeyPublic,
+                                                           @NonNull PublicKey eDeviceKeyPublic,
+                                                           @NonNull byte[] encodedSessionTranscript) {
+        return new SessionEncryption(ROLE_MDOC_READER, eReaderKeyPrivate, eDeviceKeyPublic,
+                encodedSessionTranscript, eReaderKeyPublic);
+    }
+
+    /**
+     * Creates a new {@link SessionEncryption} object with the ROLE_MDOC role.
+     *
+     * @param eDeviceKeyPrivate the device private ephemeral key.
+     * @param eReaderKeyPublic the reader public ephemeral key.
+     * @param encodedSessionTranscript the bytes of the <code>SessionTranscript</code> CBOR.
+     * @return a new {@link SessionEncryption} object with the ROLE_MDOC role.
+     */
+    public static SessionEncryption createDeviceSessionEncryption(@NonNull PrivateKey eDeviceKeyPrivate,
+                                                           @NonNull PublicKey eReaderKeyPublic,
+                                                           @NonNull byte[] encodedSessionTranscript) {
+        return new SessionEncryption(ROLE_MDOC, eDeviceKeyPrivate, eReaderKeyPublic,
+                encodedSessionTranscript, null);
+    }
```

Comment:
If we change the constructor as proposed above, I don't think we need `createReaderSessionEncryption()` or `createDeviceSessionEncryption()`.